cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/JpaRestApiClient.java

```diff
@@ -0,0 +1,294 @@
+ * authorisation is implemented in the API this class will be updated to add authentication to the
+ * requests being made.
+ */
+@Component
+public class JpaRestApiClient {
+
+  public static final String API_ROOT_PATH = "/resources/";
+  public static final String TENANT_ID_SYSTEM_PROPERTY_NAME = "cucumber.jparest.tenantId";
+  private static final String TENANT_ID_SYSTEM_PROPERTY_EL
+      = "#{systemProperties['" + TENANT_ID_SYSTEM_PROPERTY_NAME + "']}";
+
+  /**
+   * Injecting tenantId parameter as a system property to avoid verbosity in BDD features. This is a
+   * temporary workaround, as in the long term, tenantId would be part of the authenticated user
+   * object and would not have to be supplied as a request parameter
+   */
+  @Value(TENANT_ID_SYSTEM_PROPERTY_EL)
+  private String tenantId;
+
+  /**
+   * @param requestSpecification Request Specification
+   * @param persona Persona
+   * @return RequestSpecification
+   */
+  private static RequestSpecification addPersonaAuthToRequestSpecification(
+      @NonNull RequestSpecification requestSpecification, @NonNull Persona persona) {
+    String auth = persona.getAuthToken();
+    if (auth != null && !auth.isEmpty()) {
+      requestSpecification.header(HttpHeaders.AUTHORIZATION, auth);
+    }
+    return requestSpecification;
+  }
+
+  /**
+   * Retrieves the full URI called as a result of the specification.
+   *
+   * @param requestSpecification The specification to retrieve the URL from
+   * @return URI
+   */
+  private static URI getUri(@NonNull RequestSpecification requestSpecification) {
+    QueryableRequestSpecification queryRequest = SpecificationQuerier.query(requestSpecification);
+
+    String absoluteUri = queryRequest.getURI();
+    URI requestUri = null;
+    try {
+      requestUri = new URI(absoluteUri);
+    } catch (URISyntaxException e) {
+      fail(e.getMessage()); //NOSONAR
+    }
+
+    return requestUri;
+  }
+
+  @Getter
+  private final ServiceRegistry serviceRegistry;
+
+  @Autowired
+  public JpaRestApiClient(ServiceRegistry serviceRegistry) {
+    this.serviceRegistry = Objects.requireNonNull(serviceRegistry,
+        "serviceRegistry must not be null");
+  }
+
+  /**
+   * Creates resources in the specified service using the provided payload.
+   *
+   * @param persona  The persona making the request.
+   * @param service  The name of the service where the resources will be created. The service name
+   *                 must exist in the {@link ServiceRegistry}
+   * @param resource The name of the type of resource to create
+   * @param payload  The resource to create
+   * @return JpaRestApiResourceResponse
+   */
+  public JpaRestApiResourceResponse create(Persona persona, String service, String resource,
+      String payload) {
+    URI uri = getResourceUri(service, resource);
+    RequestSpecification spec = given()
+        .baseUri(uri.toString())
+        .body(payload)
+        .queryParam(TENANT_ID.getParamName(), tenantId);
```

Comment:
Just wanted to discuss the dependency this creates between cucumber-jparest and jparest. It's not the only one. `cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/stepdefinitions/CucumberConfig.java` also has a dependency to get the objectmapper config. Are there options?

Reply:
Re: the TENANT_ID request param, it is a temporary workaround anyway (see my response to your comment about hardcoded tenant id value in POM). In the meantime, we can still define a cucumber-jparest constant instead of re-using jparest's one.

Re: objectMapper config, it can also be defined in cucumber-jparest.

If this library is intended to test only projects using jparest library, then maybe the decoupling of jparest and cucumber-jparest doesn't add much value